extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,429 @@
+		$dbName = $dbw->getDBname();
+		$this->addLog( "Processing {$dbName}.{$table}.{$extra['username_column']}." );
+
+		try {
+			if ( !$dbw->tableExists( $table ) ) {
+				$this->addLog( "Table \"$table\" does not exist in database {$dbName}" );
+				$this->addWarning( wfMessage( 'coppatool-warn-table-missing', $dbName, $table )->inContentLanguage()->text() );
+
+				return false;
+			}
+
+			$values = array(
+				$extra['username_column'] => $newIP,
+			);
+
+			$conds = array(
+				$extra['username_column'] => $oldIP,
+			);
+
+			if ( !empty( $extra['conds'] ) ) {
+				$conds = array_merge( $extra['conds'], $conds );
+			}
+
+			$opts = array(
+				'LIMIT' => self::MAX_ROWS_PER_QUERY,
+			);
+
+			$affectedRows = 1;
+
+			while ( $affectedRows > 0 ) {
+				$dbw->update( $table, $values, $conds, __CLASS__ . '::' . __METHOD__, $opts );
+				$affectedRows = $dbw->affectedRows();
+				$this->addLog( "SQL: " . $dbw->lastQuery() );
+				$dbw->commit();
+				$this->addLog( "In {$dbName}.{$table}.{$extra['username_column']} {$affectedRows} row(s) was(were) updated." );
+				sleep( RENAMEIP_LOOP_PAUSE );
+			}
+		} catch ( Exception $e ) {
+			$this->addLog( "Exception in renameInTable(): " . $e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() );
+		}
+
+		$this->addLog( "Finished processing {$dbName}.{$table}.{$extra['username_column']}." );
+
+		return true;
+	}
+
+	/**
+	 * Sends the internal log message to the specified destination
+	 *
+	 * @param $text string Log message
+	 * @param $arg1 mixed Multiple format parameters
+	 */
+	public function addLog( $text, $arg1 = null ) {
+		if ( func_num_args() > 1 ) {
+			$args = func_get_args();
+			$args = array_slice( $args, 1 );
+			$text = vsprintf( $text, $args );
+		}
+		foreach ( $this->mLogDestinations as $destinationEntry ) {
+			$logDestination = $destinationEntry[0];
+
+			/** @var BatchTask $logTask */
+			$logTask = $destinationEntry[1];
+			switch ( $logDestination ) {
+				case self::LOG_BATCH_TASK:
+					$logTask->log( $text );
+					break;
+				case self::LOG_OUTPUT:
+					echo $text . "\n";
+					break;
+//				case self::LOG_STANDARD:
+				default:
+					wfDebugLog( __CLASS__, $text );
+			}
+		}
+	}
+
+	/**
+	 * Logs the message to main user-visible log
+	 *
+	 * @param string $action
+	 * @param $text string Log message
+	 */
+	public function addMainLog( $action, $text ) {
+		StaffLogger::log(
+			'coppatool',
+			$action,
+			$this->mRequestorId,
+			$this->mRequestorName,
+			$this->mUserId,
+			$this->mNewIPAddress,
+			$text
+		);
+	}
+
+	/**
+	 * Adds log message in Special:Log for the current wiki
+	 *
+	 * @param $text string Log message
+	 */
+	public function addLocalLog( $text ) {
+		$log = new LogPage( 'coppatool' );
+		$log->addEntry( 'coppatool', Title::newFromText( $this->mOldIPAddress, NS_USER ), $text, array(), User::newFromId( $this->mRequestorId ) );
+	}
+
+	public function setRequestorUser() {
+		global $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app